### PR TITLE
Feature/unsubscribe from multiple topics

### PR DESCRIPTION
**Pull request by SivaAkiro**
Motivation:

Multiple MQTT topics can be unsubscribed together. 

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/main/java/io/vertx/mqtt/MqttClient.java [hunk 1]
```diff
@@ -253,6 +254,24 @@ static MqttClient create(Vertx vertx) {
    */
   Future<Integer> unsubscribe(String topic);
 
+  /**
+   * Unsubscribe from receiving messages on given list of topic
+   *
+   * @param topics Topic you want to unsubscribe from
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/MqttClient.java:260]: I would say "list of topics you want ..."

File: src/main/java/io/vertx/mqtt/MqttClient.java [hunk 2]
```diff
@@ -253,6 +254,24 @@ static MqttClient create(Vertx vertx) {
...
+   * @return a {@code Future} completed after UNSUBSCRIBE packet sent with packetid
+   */
+  Future<Integer> unsubscribe(List<String> topics);
+
+  /**
+   * Unsubscribe from receiving messages on given topics
+   *
+   * @param topics Topics you want to unsubscribe from
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/MqttClient.java:268]: Ditto as before

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 3]
```diff
@@ -647,6 +658,23 @@ public Future<Integer> unsubscribe(String topic) {
...
+   * Unsubscribe from receiving messages on given topics
+   *
+   * @param topics                 Topics you want to unsubscribe from
+   * @param unsubscribeSentHandler handler called after UNSUBSCRIBE packet sent
+   * @return current MQTT client instance
+   */
+  @Override
+  public MqttClient unsubscribe(List<String> topics,
+    Handler<AsyncResult<Integer>> unsubscribeSentHandler) {
+    Future<Integer> fut = unsubscribe((topics));
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java:671]: too many parenthesis around `topics` ?

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 4]
```diff
@@ -1056,22 +1084,22 @@ private void handlePuback(int pubackMessageId) {
...
+        log.debug("PUBLISH expiration timer fired but QoS 1 message has already been PUBACKed by server");
+        return;
       }
     }
+    countInflightQueue--;
+    Handler<Integer> handler = publishCompletionExpirationHandler();
+    if (handler != null) {
+      handler.handle(expiredMessage.packetId);
+    }
+  }
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java:1102]: Are these changes related to this PR? It seems not, can you avoid them please?

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 5]
```diff
@@ -1101,21 +1129,21 @@ private void handlePubcomp(int pubcompMessageId) {
...
+        log.debug("PUBCOMP expiration timer fired but QoS 2 message has already been PUBCOMPed by server");
+        return;
       }
     }
+    countInflightQueue--;
+    Handler<Integer> handler = publishCompletionExpirationHandler();
+    if (handler != null) {
+      handler.handle(expiredMessage.packetId);
+    }
+  }
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java:1146]: Same as above if these changes are not related to the purpose of the PR please.

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 6]
```diff
@@ -1141,21 +1169,21 @@ private void handlePubrec(int pubrecMessageId) {
...
+        log.debug("PUBREC expiration timer fired but QoS 2 message has already been PUBRECed by server");
+        return;
       }
     }
+    countInflightQueue--;
+    Handler<Integer> handler = publishCompletionExpirationHandler();
+    if (handler != null) {
+      handler.handle(expiredMessage.packetId);
+    }
+  }
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java:1186]: Again ... not needed changes.

File: src/test/java/io/vertx/mqtt/it/MqttClientUnsubscribeIT.java [hunk 7]
```diff
@@ -36,6 +39,10 @@ public class MqttClientUnsubscribeIT extends MqttClientBaseIT {
 
   private static final String MQTT_TOPIC = "/my_topic";
 
+  private static final List<String> MQTT_TOPIC_LIST = Stream.of("my_topic1", "my_topic2").collect(
+    Collectors.toList());
```
Comment (ppatierno) [inline src/test/java/io/vertx/mqtt/it/MqttClientUnsubscribeIT.java:43]: why this is on a new line?

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java (hunk 4 above)
Comment (SivaAkiro) [reply src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java:1101]: Its the indentation issue, corrected as per the standard.

**Comment by SivaAkiro** [comment]
Fixed the review comments, can you please review?

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks for the PR!

**Review by ppatierno — APPROVED** [review]
What is changed? Approved again.

**Comment by SivaAkiro** [comment]
Nothing is changed, wrongly clicked on the refresh button.

**Review by vietj — COMMENTED** [review]
LGTM, fixed a couple of idents present in the original file
